zk-token-sdk/src/range_proof/util.rs

```diff
@@ -99,15 +99,15 @@ pub fn read32(data: &[u8]) -> [u8; 32] {
 ///    {\langle {\mathbf{a}}, {\mathbf{b}} \rangle} = \sum\_{i=0}^{n-1} a\_i \cdot b\_i.
 /// \\]
 /// Panics if the lengths of \\(\mathbf{a}\\) and \\(\mathbf{b}\\) are not equal.
-pub fn inner_product(a: &[Scalar], b: &[Scalar]) -> Scalar {
+pub fn inner_product(a: &[Scalar], b: &[Scalar]) -> Option<Scalar> {
```

Comment:
nit-up-to-you: It might be easier to have this return the `RangeProofGenerationError::InnerProductLengthMismatch` directly

Reply:
Yes, that is a good point!

Reply:
Yeah ok, that makes sense